### PR TITLE
Impl Version On Delete for Single Bucket Branch

### PR DESCRIPTION
**Pull request by bstopp**
## Description

Add the change in #86 to the Single Bucket branch.

## Motivation and Context

Bring Single Bucket into feature parity with main branch.

## How Has This Been Tested?

Added/updated tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bosschaert** [comment]
There is a small fix that we need to make atop of this: https://github.com/adobe/da-admin/pull/95

File: src/storage/object/delete.js [hunk 1]
```diff
@@ -9,35 +9,54 @@
...
+ * @param {DaCtx} daCtx the DA Context
+ * @param {String} key the key of the object to delete (excluding Org)
+ * @param {Boolean} isMove if this was initiated by a move operation
+ * @return {Promise<void>}
+ */
+export async function deleteObject(env, daCtx, key, isMove = false) {
+  const fname = key.split('/').pop();
+
+  const tmpCtx = { ...daCtx, key }; // For next calls, ctx needs the passed key, as it could contain a folder
+  if (fname.includes('.') && !key.endsWith('.props')) {
```
Comment (bosschaert) [inline src/storage/object/delete.js:28]: We should also check that fname does not _start with_ a dot, as there are hidden directories that do.

File: src/storage/object/delete.js [hunk 2]
```diff
@@ -9,35 +9,54 @@
...
+  keys.push(daCtx.key, `${daCtx.key}.props`);
   let truncated = false;
   do {
-    const r2objects = await env.DA_CONTENT.list({ prefix, limit: 500 });
+    const r2objects = await env.DA_CONTENT.list({ prefix, limit: 100 });
     const { objects } = r2objects;
     truncated = r2objects.truncated;
-    keys.push(...objects.map(({ key }) => key));
-    await env.DA_CONTENT.delete(keys);
+    keys.push(...objects.map(({ key }) => key.split('/').slice(1).join('/')));
```
Comment (bosschaert) [inline src/storage/object/delete.js:53]: Am I interpreting this correctly that it eats off the first segment of the object? I.e. the org name?

Comment (bstopp) [reply src/storage/object/delete.js:53]: Yep

File: src/storage/object/move.js [hunk 3]
```diff
@@ -53,14 +57,21 @@ export default async function moveObject(env, daCtx, details) {
...
+      sourceList.forEach(({ src, dest }) => {
+        promises.push(
+          copyFile(env, daCtx, src, dest, true)
+            .then(async (res) => {
+              if (res.success) {
+                await deleteObject(env, daCtx, src, true);
+              }
+              return res;
+            })
+            .then((res) => results.push(res)),
```
Comment (bosschaert) [inline src/storage/object/move.js:71]: It looks like the `results` aren't used, should they be?

Comment (bstopp) [reply src/storage/object/move.js:71]: Yeah, they're ignored at the moment.

**Review by bosschaert — APPROVED** [review]

**Pull request merged**